sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1498,6 +1498,45 @@ def test_neighbors_validate_parameters(Estimator):
         nbrs.predict([[]])
 
 
+@pytest.mark.parametrize(
+    "Estimator",
+    [
+        neighbors.KNeighborsClassifier,
+        neighbors.RadiusNeighborsClassifier,
+        neighbors.KNeighborsRegressor,
+        neighbors.RadiusNeighborsRegressor,
+    ],
+)
+def test_neighbors_non_metric_minkowski(Estimator):
+    """
+    Validation of all classes extending NeighborsBase for minkowski metric with p < 1
+    """
+    X = rng.random_sample((10, 2))
```

Comment:
The heuristic for setting `_fit_method` when `algo="auto"` is using `n_features`:

https://github.com/scikit-learn/scikit-learn/blob/27d3899432bef52e6706f9112cab86837497e204/sklearn/neighbors/_base.py#L585-L596

I am proposing parametrising this test accordingly to assert most possible scenarii.

```suggestion
@pytest.mark.parametrize(
    "Estimator",
    [
        neighbors.KNeighborsClassifier,
        neighbors.RadiusNeighborsClassifier,
        neighbors.KNeighborsRegressor,
        neighbors.RadiusNeighborsRegressor,
    ],
)
@pytest.mark.parametrize("n_features", [2, 100])
def test_neighbors_non_metric_minkowski(Estimator, n_features):
    """
    Validation of all classes extending NeighborsBase for minkowski metric with p < 1
    """
    X = rng.random_sample((10, n_features))
```

Reply:
parameterized tests as suggested.